sql/core/src/test/resources/sql-tests/results/double-quoted-identifiers.sql.out

```diff
@@ -277,74 +277,110 @@ SELECT 1 FROM "not_exist"
 -- !query schema
 struct<>
 -- !query output
-org.apache.spark.sql.AnalysisException
-Table or view not found: not_exist; line 1 pos 14
+org.apache.spark.sql.catalyst.parser.ParseException
```

Comment:
hmm, do we really need to run this SQL gold file in non-ansi mode? We will just see a bunch of parser errors.

Reply:
I intend to ensure it won't work when ANSI mode is off.

Reply:
I am seeing there is an extra testing here:

so this test file itself contains both `double_quoted_identifiers=false` and `double_quoted_identifiers=true` and now we run it into both ANSI and non-ANSI.

So I think what the unique testing coverage is:
1. non-ANSI and `double_quoted_identifiers=false`, so double quoted is still a string.
2. non-ANSI and `double_quoted_identifiers=true`, so we see parser exception.
3. ANSI and `double_quoted_identifiers=true`, this feature is on and being tested.

But for ANSI and `double_quoted_identifiers=false` which seems to be the same as number 2 above?

Reply:
Again, the test set is small. I don't see the downside of testing all the combinations.
Let's consider about reducing the tests when it becomes bigger, which is quite unlikely.

Reply:
I think it's time to refactor it now, as we run it again in the ansi mode tests.

It's not just unnecessary test time, but also brings confusion: why do we test `double_quoted_identifiers=false` in ansi mode?

Reply:
> but also brings confusion: why do we test double_quoted_identifiers=false in ansi mode?

Do you mean "in non-ansi" mode?

Reply:
@cloud-fan so we at least need to test the 3 cases below:
* non-ansi mode and double-quoted id enabled
* ansi mode and double-quoted id enabled
* ansi mode and double-quoted id disabled

Any suggestion on the refactor?

Reply:
We can have a `double-quoted-identifiers.sql` without duplicate queries. In `.../inputs/ansi/`, we add 2 files to import the golden file with different configs
```
// file double-quoted-identifiers-enabled.sql
--SET spark.sql.ansi.double_quoted_identifiers = true
--IMPORT double-quoted-identifiers.sql
```
```
// file double-quoted-identifiers-disabled.sql
--SET spark.sql.ansi.double_quoted_identifiers = false
--IMPORT double-quoted-identifiers.sql
```

Reply:
> non-ansi mode and double-quoted id enabled

I don't think so. This is a subconfig if ansi mode, and we only need to test the default case in non-ansi mode.

Reply:
@cloud-fan Thanks for the suggestion. I updated the tests.